service/src/main/java/io/camunda/service/license/LicenseType.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.service.license;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+public enum LicenseType {
+  SAAS("saas"),
+  PRODUCTION("production"),
+  UNKNOWN("unknown");
+  private static final Map<String, LicenseType> ENUM_MAP;
+
+  static {
+    final Map<String, LicenseType> map = new HashMap<>();
+    for (final LicenseType instance : LicenseType.values()) {
+      map.put(instance.getName().toLowerCase(), instance);
+    }
+    ENUM_MAP = Collections.unmodifiableMap(map);
+  }
+
+  private final String name;
+
+  LicenseType(final String name) {
+    this.name = name;
+  }
+
+  public String getName() {
+    return name;
+  }
+
+  public static LicenseType get(final String name) {
+    return ENUM_MAP.getOrDefault(name.toLowerCase(), LicenseType.UNKNOWN);
+  }
```

Comment:
could this not be simplified with a wrapped Enum.valueOf?

Reply:
I'm getting more safety from the `get()` method I implemented. `Enum.valueOf()` is case-sensitive, whereas the method I implemented can handle any case. The `get()` method can also handle the case of `null` and will return an `unknown` license type